client/components/plans/plan-list/index.jsx

```diff
@@ -25,29 +27,47 @@ module.exports = React.createClass( {
 	render: function() {
 		var plans = this.props.plans,
 			showJetpackPlans = false,
+			isLoadingSitePlans = ! this.props.isInSignup && ! this.props.sitePlans.hasLoadedFromServer,
 			site,
-			plansList;
+			plansList,
+			currentPlan;
+
+		if ( plans.length === 0 || isLoadingSitePlans ) {
+			plansList = times( 3, function( n ) {
+				return (
+					<Plan placeholder={ true }
+						isInSignup={ this.props.isInSignup }
+						key={ 'plan-' + n } />
+				);
+			}, this );
+
+			return (
+				<div className="plans-list">{ plansList }</div>
+			);
+		}
 
 		if ( this.props.sites ) {
 			site = this.props.sites.getSelectedSite();
 			showJetpackPlans = site && site.jetpack;
 		}
 
-		// check if this site was registered via the JPPHP "Jetpack Start" program
-		// if so, we want to display a message that this plan is managed via the hosting partner
-
-		if ( this.props.siteSpecificPlansDetailsList && this.props.siteSpecificPlansDetailsList.hasJpphpBundle( site.domain )  ) {
-			return (
-				<Card>
-					<p>
-						{
-							this.translate( 'This plan is managed by your web host. ' +
-								'Please log into your host\'s control panel to manage subscription ' +
-								'and billing information.' )
-						}
-					</p>
-				</Card>
-			);
+		if ( ! this.props.isInSignup ) {
+			// check if this site was registered via the JPPHP "Jetpack Start" program
+			// if so, we want to display a message that this plan is managed via the hosting partner
+			currentPlan = find( this.props.sitePlans.data, { currentPlan: true } );
```

Comment:
I wonder if it would not be the time to introduce a function to retrieve the current plan in the [`plans` module](https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/index.js). What do you think?